protostar/starknet/cheatable_cached_state.py

```diff
@@ -122,3 +113,42 @@ def get_abi_from_contract_address(self, contract_address: int) -> AbiType:
             )
 
         return self.class_hash_to_contract_abi_map[class_hash]
+
+
+def cheaters_of(state: StateProxy) -> Cheaters:
+    """
+    Extracts the ``Cheaters`` object from any State structures.
+
+    This function workarounds limitations of the inheritance design of ``State`` classes family,
+    preventing us from exposing the `cheaters` field via state interface classes like ``SyncState``.
+    """
```

Comment:
does the syscall handler always interact with one of those, and not just one specific implementation that we use? i'm kind of missing that part

Reply:
Syscall handler always interacts with a thing typed as `SyncState` which is an abstract interface implemented by arbitrary classes. Ideally I would like to add a method to `SyncState` but I cannot do this because I cannot modify a defined type nor define an extension to it in Python 😅 

I also avoid making assumptions that syscalls always get an instance of `StateSyncifier`, because this fact may change in the future.

Reply:
Extension methods [are theoretically possible](https://stackoverflow.com/a/7139161), but I think both @kasperski95 and Pyright will not like it 😆